src/datasets/open_weiboscope.py

```diff
@@ -0,0 +1,185 @@
+            (string)
+        form -- return results in hanzi, pinyin romanization, or one-hot encodings of romanization?
+            can take values of 'hanzi', 'pinyin', or 'onehot' (string)
+        train_pct -- what percent of dataset should go to training (remainder goes to test)?  (float)
+        rng_seed -- value for seeding random number generator
+        min_length -- enforce a minimum length, in characters, for the 
+            dataset? Counted in hanzi for form='hanzi' and in roman characters 
+            for form='pinyin'. Texts that are too short will be excluded. (int)
+        max_length -- enforce a maximum length, in characters, for the dataset?
+            Counted in hanzi or roman characters as approriate (see above).
+            Texts that are too long will be truncated at the end. (int)
+        pad_out -- for texts shorter than max_length, should they be padded out
+            at the end with spaces?
+
+    @Return:
+        a generator over a tuples of review text (unicode or numpy array) and whether or not 
+        the tweet was deleted (bool)
+
+    """
+
+    if not os.path.exists(file_path):
+        # download repository files and unzip them
+        os.makedirs(file_path)
+        for remote_path in [ "http://weiboscope.jmsc.hku.hk/datazip/week{}.zip".format(a) for a in [ str(b) for b in range(1, 52) ] ]:
+            local_zip = get_file(remote_path, file_path)
+            with ZipFile(local_zip) as zf:
+                zf.extractall(file_path)
+
+    # get list of weekNN.csv files at file_path
+    ow_files = [ os.path.join(file_path, f) for f in os.listdir(file_path) if re.match(r"week[0-9]{,2}\.csv", f) is not None ]
+    assert ow_files is not []
+    
+    # strategy: randomize order of weeks (individual files), sample in order from each week.
+    try:
+        random.seed(rng_seed)
+    except:
+        pass
+    random.shuffle(ow_files)
+    split_on = int(len(ow_files) * train_pct)
+    data_sets = {}
+    logger.debug("Shuffle order: {}, split on {}".format(ow_files, split_on))
+    data_sets['train'], data_sets['test'] = ow_files[:split_on], ow_files[split_on:]
+    logger.debug(data_sets)
+    for table_path in data_sets[which_set]:
+        with codecs.open(table_path, "r", encoding="utf-8") as f:
+            logging.debug("In file {}".format(table_path))
+            for line in unicode_csv_reader(f):
+                try:
+                    records_split = line
+                    if len(records_split) != 11:
+                        raise  BadRecordException("Comma split error on mid={} in"
+                                         "file {} (len of record: {})".format(
+                                            records_split[0], 
+                                            os.path.basename(table_path),
+                                            len(records_split)))
+            
+                    # text on field 6, deleted on field 9. screen out retweets (field 1) 
```

Comment:
this is a good comment; it might help if you explicitly specify these into variables (i.e. index_text = 6) so you're not hard-coding those array indices that could change in the future